File: Algorithm/Programmers/Python/Level 1/모의고사.py
def solution(answers):
    a = [1, 2, 3, 4, 5]
    b = [2, 1, 2, 3, 2, 4, 2, 5]
    c = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    tmp = [0, 0, 0]
    answer = []
    for i in range(0, len(answers)):
        if answers[i] == a[i % 5]:
            tmp[0] += 1

        if answers[i] == b[i % 8]:
            tmp[1] += 1

        if answers[i] == c[i % 10]:
            tmp[2] += 1

    m = max(tmp)

    for i in range(3):
        if tmp[i] == m:
            answer.append(i + 1)
    return answer

File: Algorithm/Programmers/Java/Level 1/모의고사.java
import java.util.*;
public class 모의고사 {
    public int[] solution(int[] answers) {
        int[] a={1,2,3,4,5};
        int[] b={2, 1, 2, 3, 2, 4, 2, 5};
        int[] c={3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] rec={0,0,0};
        for(int i=0; i<answers.length; i++){
            int correct=answers[i];
            
            if (a[i%5]==correct){
                rec[0]+=1;
            }
            if (b[i%8]==correct){
                rec[1]+=1;
            }
            if (c[i%10]==correct){
                rec[2]+=1;
            }            
        }
        
        int m=0;
        for (int r:rec){
            m=Math.max(m,r);
        }
        
        ArrayList<Integer> tmp = new ArrayList();
        for (int i=0; i<rec.length; i++){
            if (rec[i]==m){
                tmp.add(i+1);
            }
        }
        int[] answer=new int[tmp.size()];
        int index=0;
        for(Integer i:tmp){
            answer[index++]=i;
        }
        return answer;
    }
}
